scalajslib/src/mill/scalajslib/worker/ScalaJSWorker.scala

```diff
@@ -12,14 +12,15 @@ import mill.{Agg, T}
 private[scalajslib] class ScalaJSWorker extends AutoCloseable {
   private var scalaJSWorkerInstanceCache = Option.empty[(Long, workerApi.ScalaJSWorkerApi)]
 
-  private def bridge(toolsClasspath: Agg[os.Path])(implicit ctx: Ctx.Home) = {
-    val classloaderSig =
-      toolsClasspath.map(p => p.toString().hashCode + os.mtime(p)).sum
+  private def bridge(toolsClasspath: Agg[mill.PathRef])(implicit ctx: Ctx.Home) = {
+    // toolsClasspath is stable and made of external libraries which have the versions
+    // in the file names. So we use the paths string to hash since it's ~50x faster
+    val classloaderSig = toolsClasspath.iterator.map(p => p.path.hashCode).sum
```

Comment:
I think using the `hashCode` of the `PathRef` is a better option. Using only the name is not guaranteed to be stable. E.g. you could dynamically download to a same local path `T.dest / "out.jar"` and would miss any change when this jar changes later.